forgerock-auth/src/main/java/org/forgerock/android/auth/WebAuthnDataRepository.kt

```diff
@@ -119,20 +119,36 @@ open class WebAuthnDataRepository internal constructor(val context: Context,
         } ?: emptyList()
     }
 
-    fun deleteByUserIdAndRpId(userId: ByteArray, rpId: String) {
+    /**
+     * Delete the provided [PublicKeyCredentialSource]
+     * @param publicKeyCredentialSource The [PublicKeyCredentialSource] to be deleted
+     */
+    fun delete(publicKeyCredentialSource: PublicKeyCredentialSource) {
         if (!this::dataRepository.isInitialized) {
             warn(TAG, "UsernameLess cannot be supported. No credential will be stored")
             return
         }
 
-        val storedCredentials = dataRepository.getStringSet(rpId, null)?.filter {
-            fromJson(JSONObject(it)).userHandle.contentEquals(userId).not()
-        }?.toSet()
+        val storedCredentials =
+            getPublicKeyCredentialSource(publicKeyCredentialSource.rpid).filter {
+                it.id.contentEquals(publicKeyCredentialSource.id).not()
+            }.map { it.toJson().toString() }.toSet()
 
-        storedCredentials?.let {
-            dataRepository.edit().putStringSet(rpId, it).apply()
+        storedCredentials.let {
```

Comment:
dont need this let { } .  Let can be only used for optional. just go straight dataRepository.edit().putStringSet(publicKeyCredentialSource.rpid, storedCredentials).apply()

Reply:
Updated